hbase-shaded/pom.xml

```diff
@@ -51,15 +51,14 @@
       <dependency>
          <groupId>org.apache.hbase</groupId>
          <artifactId>hbase-resource-bundle</artifactId>
-         <version>${project.version}</version>
          <optional>true</optional>
       </dependency>
-      <!-- log4j has to be non-optional for Hadoop 2 atleast -->
+      <!-- put the log implementations to optional -->
```

Comment:
At least for hadoop 2.8+, they have already moved to slf4j.

Reply:
After trying HBASE-19577, some hadoop classes do have dependencies on log4j directly, for example the FSNamesystem class. But usually we do not need to touch this class unless in UT. Anyway, since we are still on log4j after this patch so I think it is fine for now. When switching to log4j2, we need to more testing to find out the direct dependencies on log4j from hadoop and try to fix them.